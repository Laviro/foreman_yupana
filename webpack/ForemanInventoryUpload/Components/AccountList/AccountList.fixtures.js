import { noop } from 'patternfly-react';

export const accounts = {
  Account1: {
    label: 'test_org1',
    upload_report_status: 'running',
    generate_report_status: 'running',
  },
  Account2: {
    label: 'test_org2',
    upload_report_status: 'unknown',
    generate_report_status: 'failure',
  },
  Account3: {
    label: 'test_org3',
    upload_report_status: 'success',
    generate_report_status: 'running',
  },
};

export const accountIDs = Object.keys(accounts);

export const API_SUCCESS_RESPONSE = accounts;

export const pollingProcessID = 0;

export const error = 'some-error';

export const accountID = 'user@redhat.com';

export const processStatusName = 'upload_report_status';

export const autoUploadEnabled = true;

export const hostObfuscationEnabled = true;

export const filterTerm = 'some_filter';

export const props = {
  accounts,
  fetchAccountsStatus: noop,
  startAccountStatusPolling: noop,
  stopAccountStatusPolling: noop,
  pollingProcessID,
};

export const fetchAccountsStatusResponse = {
  data: { accounts, autoUploadEnabled, hostObfuscationEnabled },
};
