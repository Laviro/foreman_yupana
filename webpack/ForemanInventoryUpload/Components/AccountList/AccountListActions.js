import API from 'foremanReact/API';
import { inventoryUrl } from '../../ForemanInventoryHelpers';
import {
  INVENTORY_ACCOUNT_STATUS_POLLING,
  INVENTORY_ACCOUNT_STATUS_POLLING_ERROR,
  INVENTORY_ACCOUNT_STATUS_POLLING_START,
  INVENTORY_ACCOUNT_STATUS_POLLING_STOP,
  INVENTORY_PROCESS_RESTART,
} from './AccountListConstants';

export const fetchAccountsStatus = () => async dispatch => {
  try {
    const {
      data: { accounts, autoUploadEnabled, hostObfuscationEnabled },
    } = await API.get(inventoryUrl('accounts'));
    dispatch({
      type: INVENTORY_ACCOUNT_STATUS_POLLING,
      payload: {
        accounts,
        autoUploadEnabled,
        hostObfuscationEnabled,
      },
    });
  } catch (error) {
    dispatch({
      type: INVENTORY_ACCOUNT_STATUS_POLLING_ERROR,
      payload: {
        error: error.message,
      },
    });
  }
};

export const startAccountStatusPolling = pollingProcessID => ({
  type: INVENTORY_ACCOUNT_STATUS_POLLING_START,
  payload: {
    pollingProcessID,
  },
});

export const stopAccountStatusPolling = pollingProcessID => dispatch => {
  clearInterval(pollingProcessID);
  dispatch({
    type: INVENTORY_ACCOUNT_STATUS_POLLING_STOP,
  });
};

export const restartProcess = (accountID, activeTab) => dispatch => {
  let processController = null;
  let processStatusName = null;

  if (activeTab === 'uploading') {
    processController = 'uploads';
    processStatusName = 'upload_report_status';
  } else {
    processController = 'reports';
    processStatusName = 'generate_report_status';
  }

  API.post(inventoryUrl(`${accountID}/${processController}`));
  dispatch({
    type: INVENTORY_PROCESS_RESTART,
    payload: {
      accountID,
      processStatusName,
    },
  });
};
