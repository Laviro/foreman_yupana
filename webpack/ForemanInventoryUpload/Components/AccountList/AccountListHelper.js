export const filterAccounts = (accounts, accountIds, filterTerm) => {
  if (!filterTerm || !accountIds.length) {
    return accountIds;
  }

  const filterTermLowerCased = filterTerm.toLowerCase();
  return accountIds.filter(id =>
    accounts[id].label.toLowerCase().includes(filterTermLowerCased)
  );
};
