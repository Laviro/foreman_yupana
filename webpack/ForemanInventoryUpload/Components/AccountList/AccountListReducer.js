import Immutable from 'seamless-immutable';
import {
  INVENTORY_ACCOUNT_STATUS_POLLING,
  INVENTORY_ACCOUNT_STATUS_POLLING_ERROR,
  INVENTORY_ACCOUNT_STATUS_POLLING_START,
  INVENTORY_PROCESS_RESTART,
} from './AccountListConstants';

const initialState = Immutable({
  accounts: {},
  pollingProcessID: 0,
  error: null,
});

export default (state = initialState, action) => {
  const {
    payload: {
      pollingProcessID,
      error,
      accounts,
      accountID,
      processStatusName,
      cloudToken,
      CloudConnectorStatus,
    } = {},
  } = action;

  switch (action.type) {
    case INVENTORY_ACCOUNT_STATUS_POLLING:
      return state.merge({
        ...state,
        accounts,
        cloudToken,
        CloudConnectorStatus,
        error: null,
      });
    case INVENTORY_ACCOUNT_STATUS_POLLING_ERROR:
      return state.merge({
        ...state,
        accounts: {},
        error,
      });
    case INVENTORY_ACCOUNT_STATUS_POLLING_START:
      return state.merge({
        ...state,
        pollingProcessID,
      });
    case INVENTORY_PROCESS_RESTART:
      return state.setIn(['accounts'], {
        ...state.accounts,
        [accountID]: {
          ...state.accounts[accountID],
          [processStatusName]: 'Restarting...',
        },
      });
    default:
      return state;
  }
};
