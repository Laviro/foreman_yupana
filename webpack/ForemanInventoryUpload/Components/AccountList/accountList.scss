.list-view-pf-expand {
  margin-top: 25px;
}
