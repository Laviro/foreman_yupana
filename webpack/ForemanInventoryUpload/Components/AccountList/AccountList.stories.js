import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { Provider } from 'react-redux';
import configureStore from '../../../stories/configureStore';
import AccountList from './AccountList';

const store = configureStore();

storiesOf('Account list', module)
  .addDecorator(withKnobs)
  .add('Account list', () => (
    <div style={{ margin: '20px' }}>
      <Provider store={store}>
        <AccountList />
      </Provider>
    </div>
  ));
