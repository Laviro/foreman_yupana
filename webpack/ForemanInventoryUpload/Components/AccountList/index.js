import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './AccountListActions';
import reducer from './AccountListReducer';
import AccountList from './AccountList';
import {
  selectAccounts,
  selectPollingProcessID,
  selectError,
} from './AccountListSelectors';
import { selectFilterTerm } from '../InventoryFilter/InventoryFilterSelectors';

// map state to props
const mapStateToProps = state => ({
  accounts: selectAccounts(state),
  pollingProcessID: selectPollingProcessID(state),
  error: selectError(state),
  filterTerm: selectFilterTerm(state),
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = { accountsList: reducer };

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(AccountList);
