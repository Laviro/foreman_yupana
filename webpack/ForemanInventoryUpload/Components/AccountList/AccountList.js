import React, { Component } from 'react';
import { ListView, noop } from 'patternfly-react';
import PropTypes from 'prop-types';
import ListItem from './Components/ListItem';
import EmptyState from './Components/EmptyState';
import ErrorState from './Components/ErrorState';
import './accountList.scss';

class AccountList extends Component {
  componentDidMount() {
    const { fetchAccountsStatus, startAccountStatusPolling } = this.props;
    fetchAccountsStatus();
    const pollingProcessID = setInterval(fetchAccountsStatus, 5000);
    startAccountStatusPolling(pollingProcessID);
  }

  componentWillUnmount() {
    const { stopAccountStatusPolling, pollingProcessID } = this.props;
    stopAccountStatusPolling(pollingProcessID);
  }

  render() {
    const { statuses, error } = this.props;
    const accountNames = Object.keys(statuses);

    if (error) {
      return <ErrorState error={error} />;
    }

    if (accountNames.length === 0) {
      return <EmptyState />;
    }
    const items = accountNames.map((name, index) => {
      const status = statuses[name];
      return (
        <ListItem
          key={index}
          name={name}
          statuses={status}
          initExpanded={index === 0}
        />
      );
    });
    return <ListView className="account_list">{items}</ListView>;
  }
}

AccountList.propTypes = {
  fetchAccountsStatus: PropTypes.func,
  startAccountStatusPolling: PropTypes.func,
  stopAccountStatusPolling: PropTypes.func,
  pollingProcessID: PropTypes.number,
  statuses: PropTypes.shape({
    generate_report_status: PropTypes.string,
    upload_report_status: PropTypes.string,
  }),
  error: PropTypes.string,
};

AccountList.defaultProps = {
  fetchAccountsStatus: noop,
  startAccountStatusPolling: noop,
  stopAccountStatusPolling: noop,
  pollingProcessID: 0,
  statuses: {
    generate_report_status: 'unknown',
    upload_report_status: 'unknown',
  },
  error: '',
};

export default AccountList;
