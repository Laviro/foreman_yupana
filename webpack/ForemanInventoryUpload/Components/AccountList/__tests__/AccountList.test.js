import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import AccountList from '../AccountList';
import { props } from '../AccountList.fixtures';

const fixtures = {
  'render with props': props,
  'show empty results': { ...props, filterTerm: 'not_matching_term' },
};

describe('AccountList', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(AccountList, fixtures));
});
