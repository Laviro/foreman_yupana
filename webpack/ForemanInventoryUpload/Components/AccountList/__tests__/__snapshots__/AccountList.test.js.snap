// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList rendering render with props 1`] = `
<ListView
  className="account_list"
>
  <ListItem
    account={
      Object {
        "generate_report_status": "running",
        "label": "test_org1",
        "upload_report_status": "running",
      }
    }
    accountID="Account1"
    key="0"
  />
  <ListItem
    account={
      Object {
        "generate_report_status": "failure",
        "label": "test_org2",
        "upload_report_status": "unknown",
      }
    }
    accountID="Account2"
    key="1"
  />
  <ListItem
    account={
      Object {
        "generate_report_status": "running",
        "label": "test_org3",
        "upload_report_status": "success",
      }
    }
    accountID="Account3"
    key="2"
  />
</ListView>
`;

exports[`AccountList rendering show empty results 1`] = `<inventoryEmptyResults />`;
